### PR TITLE
New Code Example: Fine-tuning RoBERTa for Sentiment Span Extraction using Hugging Face Transformers

### PR DESCRIPTION
**Pull request by Sreyan88**
This code example shows how to fine-tune RoBERTa for sentiment span extraction using Hugging Face Transformers. Additionally, it also shows best practices for Hugging Face Datasets.

This is part of my GSoC 2022 contribution. More information about my project can be found on [this link](https://docs.google.com/document/d/1fLDLwIhnwDUz3uUV8RyUZiOlmTN9Uzy5ZuvI8iDDFf8/edit#heading=h.bvepijofrmt9) and [this link](https://summerofcode.withgoogle.com/programs/2022/projects/tLgwQubw)

Thank You!

### CODE REVIEW

File: examples/nlp/sentiment_span_extraction.py [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+"""
+
+"""
+### Importing the necessary libraries
+"""
+
+import random
+import logging
+
+import torch
```
Comment (fchollet) [inline examples/nlp/sentiment_span_extraction.py:59]: This import appears to be unused.

File: examples/nlp/sentiment_span_extraction.py [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+pip install git+https://github.com/huggingface/transformers.git
+pip install datasets
+pip install huggingface-hub
+"""
+
+"""
+### Importing the necessary libraries
+"""
+
+import random
```
Comment (fchollet) [inline examples/nlp/sentiment_span_extraction.py:56]: Since this example uses randomness, consider setting the random seed.

File: examples/nlp/sentiment_span_extraction.py [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+
+MODEL_CHECKPOINT = "roberta-base"
+BASE_URL = "https://huggingface.co/datasets/SetFit/tweet_sentiment_extraction/raw/main/raw_data/"
+
+"""
+## Load the Kaggle Tweet Sentiment Extraction Dataset
+"""
+
+"""
+We will now download the [Kaggle Tweet Sentiment Extraction
```
Comment (fchollet) [inline examples/nlp/sentiment_span_extraction.py:88]: Make sure all links fit on a single line.

File: examples/nlp/sentiment_span_extraction.py [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+span to extract from the sentence. In this notebook, we just append the original sentiment
+label while training and inference.
+
+Let's see a random example of how our input to the model will look like. We will write a
+simple function `show_random_example()` to do that:
+"""
+
+
+def show_random_example():
+
```
Comment (fchollet) [inline examples/nlp/sentiment_span_extraction.py:173]: Remove blank line.

File: examples/nlp/sentiment_span_extraction.py [hunk 5]
```diff
@@ -0,0 +1,522 @@
...
+Next, let's tokenize the sentiment labels in our dataset for use in data preparation.
+"""
+
+sentiment_id = {}
+
+for sentiment in list(set(dataset["train"]["sentiment"])):
+    sentiment_id[sentiment] = tokenizer.convert_tokens_to_ids("Ġ" + sentiment)
+
+sentiment_id
```
Comment (fchollet) [inline examples/nlp/sentiment_span_extraction.py:198]: Use `print()`.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

**Comment by pcoet** [comment]
@Sreyan88 Thanks for the PR. Are you still waiting for a review? Just want to make sure we didn't miss this. Thanks!

**Comment by Sreyan88** [comment]
Hi @pcoet , All the latest comments are addressed. Thank You!

**Comment by pcoet** [comment]
Thanks, @Sreyan88! It looks like you made all of @fchollet's requested changes. Can you go ahead and generate the notebook and markdown files? https://github.com/keras-team/keras-io#creating-a-new-example-starting-from-a-python-script
